pkg/services/cosmosdb/common-update.go

```diff
@@ -15,3 +17,66 @@ func (
 ) GetUpdater(service.Plan) (service.Updater, error) {
 	return service.NewUpdater()
 }
+
+// This function will build a map in which only read regions changed.
+func (c *cosmosAccountManager) buildGoTemplateParamsOnlyRegionChanged(
+	provisioningParameters *service.ProvisioningParameters,
+	updatingParameters *service.ProvisioningParameters,
+	dt *cosmosdbInstanceDetails,
+	kind string,
+) (map[string]interface{}, error) {
+	p := map[string]interface{}{}
+	p["name"] = dt.DatabaseAccountName
+	p["kind"] = kind
+	p["location"] = provisioningParameters.GetString("location")
+	p["readRegions"] = updatingParameters.GetStringArray("readLocations")
+	if provisioningParameters.GetString("autoFailoverEnabled") == "enabled" {
+		p["enableAutomaticFailover"] = true
+	} else {
+		p["enableAutomaticFailover"] = false
+	}
+
+	filters := []string{}
+	ipFilters := provisioningParameters.GetObject("ipFilters")
+	if ipFilters.GetString("allowAzure") != disabled {
+		filters = append(filters, "0.0.0.0")
```

Comment:
this is the same as the last else.
suppose should not be the same?
and "0.0.0.0" seems a bit dangerous.

Reply:
This function was pasted from [common-provision.go](https://github.com/norshtein/open-service-broker-azure/blob/region-in-template/pkg/services/cosmosdb/common-provision.go#L47). According to the [doc](https://docs.microsoft.com/en-us/azure/cosmos-db/firewall-support#connections-from-global-azure-datacenters-or-azure-paas-services), ip address "0.0.0.0" denotes access from Azure services, not the same as traditional "allow all ipaddress". And agree with you it's very strange that the statement is the same as the last else. I also don't know why they write in this way...

Reply:
Add on: this function was pasted, with [locations read from updating parameters](https://github.com/norshtein/open-service-broker-azure/blob/region-in-template/pkg/services/cosmosdb/common-update.go#L32) rather than provision parameters. Except this, everything is the same.

Reply:
ok, and why
 L42 filters = append(filters, "0.0.0.0")
is same with
L 72 filters = append(filters, "0.0.0.0")
maybe we should merge them instead of in if and else.